src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Escaping.cs

```diff
@@ -212,304 +216,28 @@ private static void EscapeNextBytes(byte value, Span<byte> destination, ref int
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Is this really needed? I assume the JIT will inline this method without the attribute too.
Also on L222 (method `IsAsciiValue(char)` the attribute could be removed.

Reply:
I applied the attribute because it was present in the other `IsAsciiValue` method.

I verified locally (Windows release build) that the IsAsciiValue methods are inlined without the attribute.

However there are many (60+) uses of AggressiveInlining in System.Text.Json similar to this -- some with a few lines of code (so not as obvious) but also some one-liners, so I'm reluctant to change without some sort of a plan or guidelines --  @ahsonkhan thoughts on when to include AggressiveInlining?

Reply:
Generally speaking, AggressiveInlining should be used sparingly and only if there is evidence it is necessary and is improving perf. I may have used it too liberally in the past, and if a method is getting inlined without the attribute, we shouldn't add one, especially for new code/changes.

**So, I wouldn't add one here.**

We can consider removing some of the other ones that might have been unnecessary, separately.

Reply:
It occurs to me that with a slightly nonstandard jit, we can flag cases where `AggressiveInlining` is redundant. In conjunction with PMI (from jitutils) we should be able to spot the majority of these cases in CoreFx and CoreCLR.

The jit currently will bypass profitability analysis when it sees `AggressiveInlining`. So the first nonstandard bit would be to always run profitability analysis and then flag cases that are both profitable and marked. The second would be to surface the results somehow... we can be low-tech here I suppose, and just write to stdout or stderr.

Profitability is contextual (depends on caller and call site) so we'd have to look at what happens at all the sites that invoke a method; if most or all of these would inline without the attribute, then we can arguably remove it.

Reply:
I will remove `AggressiveInlining` for the methods in this file except for this one that still requires it (not inlined by default):
`private static bool NeedsEscaping(char value) => value > LastAsciiCharacter || AllowList[value] == 0;`